crates/moonbuild/src/dry_run.rs

```diff
@@ -98,15 +98,26 @@ pub fn print_run_commands(
                 watfile = watfile.replacen(&source_dir.display().to_string(), ".", 1);
             }
 
-            let mut moonrun_command = if let Some(cmd) = cmd {
+            let moonrun_command = if let Some(cmd) = cmd {
                 let cmd = cmd.display();
-                format!("{cmd} {watfile} --")
-            } else {
+                if args.is_empty() {
+                    format!("{cmd} {watfile}")
+                } else {
+                    // Only Wasm/WasmGC backends need the -- separator for moonrun
+                    match target_backend {
+                        TargetBackend::Wasm | TargetBackend::WasmGC => {
+                            format!("{cmd} {watfile} -- {}", args.join(" "))
+                        }
+                        _ => {
+                            format!("{cmd} {watfile} {}", args.join(" "))
+                        }
+                    }
+                }
+            } else if args.is_empty() {
                 watfile
+            } else {
+                format!("{watfile} {}", args.join(" "))
             };
-            if !args.is_empty() {
-                moonrun_command = format!("{moonrun_command} {}", args.join(" "));
-            }
 
             println!("{moonrun_command}");
```

Comment:
Ideally it might be preferred to just join the args in a vec and use `shlex` for it? Don't know if it's worth the trouble